flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1898,8 +1898,8 @@ class ScalarExprLowering {
     bool addHostAssociations = false;
     if (const auto *sym = caller.getIfIndirectCallSymbol()) {
       funcPointer = symMap.lookupSymbol(*sym).getAddr();
-      assert(funcPointer &&
-             "dummy procedure or procedure pointer not in symbol map");
+      if (!funcPointer)
+        TODO(loc, "calling a procedure declared in an outer scope");
```

Comment:
Should this be " a dummy procedure" rather than "a procedure" ? I think the assert is only firing for host procedure dummy procedure arguments called inside an internal procedure, so "a procedure "may be a bit too vague.

Reply:
> Should this be " a dummy procedure" rather than "a procedure" ? I think the assert is only firing for host procedure dummy procedure arguments called inside an internal procedure, so "a procedure "may be a bit too vague.

Jean, you're a genius!  I'll make that change.